spec/features/complexity_level_feature_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+feature 'complexity level feature' do
+  let(:offender) { build(:nomis_offender, complexityLevel: 'high', agencyId: womens_prison.code, firstName: 'Sally', lastName: 'Albright') }
+  let(:womens_prison) { build(:womens_prison) }
+  let(:offenders) { [offender] }
+  let(:pom) { build(:pom) }
+  let(:spo) { build(:pom) }
+  let(:test_strategy) { Flipflop::FeatureSet.current.test! }
+  let(:offender_no) { offender.fetch(:offenderNo) }
+
+
+  before do
+    create(:allocation, nomis_offender_id: offender.fetch(:offenderNo), primary_pom_nomis_id: pom.staff_id,  prison: womens_prison.code)
+    create(:case_information, nomis_offender_id: offender.fetch(:offenderNo))
+    expect(HmppsApi::ComplexityApi).to receive(:save).with(offender.fetch(:offenderNo), level: 'low', username: "MOIC_POM", reason: 'bla bla bla')
+
+    stub_offenders_for_prison(womens_prison.code, offenders)
+    stub_signin_spo(spo, [womens_prison.code])
+    stub_offender(offender)
+    stub_poms(womens_prison.code, [pom, spo])
+    stub_keyworker(womens_prison.code, offender.fetch(:offenderNo), build(:keyworker))
+    test_strategy.switch!(:womens_estate, true)
+  end
+
+  after do
+    test_strategy.switch!(:womens_estate, false)
+  end
+
+  it 'update the complexity level journey', :js do
+    # Journey starts on prisoner profile page. It should display the existing complexity level and a change link.
+    visit  prison_allocation_path(nomis_offender_id: offender.fetch(:offenderNo), prison_id: womens_prison.code)
+
+    expect(page).to have_text('Complexity level')
+    expect(page).to have_css('#complexity-level', text: 'High')
+    expect(page).to have_css('#complexity-badge', text: 'HIGH COMPLEXITY')
+
+    within(:css, "td#complexity-level") do
+      click_link('Change')
+    end
+
+    # Taken to the edit page to update the complexity level
+    expect(page).to have_text 'Update complexity of need level'
+
+    # The radio buttons should have the current complexity level checked
+    high_radio_button = find('#complexity-level-high-field', visible: false)
+    expect(high_radio_button).to be_checked
+
+    # Updated level radio button checked
+    find('label[for=complexity-level-low-field]').click
+    find('#update', visible: false)
+
+    # Text area left blank
+    click_on('Update')
+    # Shows errors
+    expect(page).to have_css('.govuk-error-message')
+    expect(page).to have_text('Enter the reason why the level has changed')
+
+    # Happy path - Textarea Filled in and complexity updated from high to low
+    # find('label[for=complexity-level-low-field]').click
+    find('textarea#complexity-reason-field-error', visible: false)
```

Comment:
This find doesn't seem to be required

Reply:
This one fails if I take it out, as later on, when I check to see that the page displays 'Low' in the confirmation page.